pkg/models/commandvalue.go

```diff
@@ -30,632 +26,593 @@ const (
-	err := binary.Read(reader, binary.BigEndian, value)
-	return err
+func NewBinaryValue(deviceResourceName string, value []byte) (*CommandValue, errors.EdgeX) {
+	return NewCommandValue(deviceResourceName, v2.ValueTypeBinary, value)
 }
 
 // ValueToString returns the string format of the value.
-// In EdgeX, float value has two kinds of representation, Base64, and eNotation.
-// Users can specify the floatEncoding in the properties value of the device profile, like floatEncoding: "Base64" or floatEncoding: "eNotation".
-func (cv *CommandValue) ValueToString(encoding ...string) (str string) {
-	if cv.Type == v2.ValueTypeString {
-		str = cv.stringValue
-		return
-	}
-
-	reader := bytes.NewReader(cv.NumericValue)
-
-	switch cv.Type {
-	case v2.ValueTypeBool:
-		var res bool
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatBool(res)
-	case v2.ValueTypeUint8:
-		var res uint8
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatUint(uint64(res), 10)
-	case v2.ValueTypeUint16:
-		var res uint16
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatUint(uint64(res), 10)
-	case v2.ValueTypeUint32:
-		var res uint32
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatUint(uint64(res), 10)
-	case v2.ValueTypeUint64:
-		var res uint64
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatUint(res, 10)
-	case v2.ValueTypeInt8:
-		var res int8
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatInt(int64(res), 10)
-	case v2.ValueTypeInt16:
-		var res int16
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatInt(int64(res), 10)
-	case v2.ValueTypeInt32:
-		var res int32
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatInt(int64(res), 10)
-	case v2.ValueTypeInt64:
-		var res int64
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatInt(res, 10)
-	case v2.ValueTypeFloat32:
-		var res float32
-		binary.Read(reader, binary.BigEndian, &res)
-		str = fmt.Sprintf("%e", res)
-	case v2.ValueTypeFloat64:
-		var res float64
-		binary.Read(reader, binary.BigEndian, &res)
-		str = fmt.Sprintf("%e", res)
-	case v2.ValueTypeBinary:
-		// produce string representation of first 20 bytes of binary value
-		str = fmt.Sprintf(fmt.Sprintf("Binary: [%v...]", string(cv.BinValue[:20])))
-	default:
-		// ArrayType
-		str = cv.stringValue
-	}
-
-	return
+func (cv *CommandValue) ValueToString() string {
```

Comment:
Why/where is this needed?

Reply:
in `transformer` package, especially doing assertion or mapping for the reading
https://github.com/edgexfoundry/device-sdk-go/blob/62dc0a4d0d8d70c39ca7894a25731db064e1c9c7/internal/transformer/transformresult.go#L606

Reply:
I don't follow the purpose of CheckAssertion, but I don't think this will product the same string values that the previous code did. The new Reading factory methods are doing this in same way, so I think you could rearrange when CheckAssertion is called so that it can be changed to use the Reading.Value rather than call this function on CommandValue.  Binary value is a special case that will need to use `str = fmt.Sprintf(fmt.Sprintf("Binary: [%v...]", string(cv.BinValue[:20])))` from old code above.

Reply:
updated for Binary type.
I prefer do the assertion/mapping with CommandValue to be aligned with the [adr](https://github.com/edgexfoundry/edgex-docs/blob/master/docs_src/design/adr/device-service/0011-DeviceService-Rest-API.md#assertions-and-mappings) 

for the other type I think there's no much difference. 
(space-separated or comma-separated for array type is the difference, but assertion and mapping should not be applied on array type)